src/js/node/zlib.ts

```diff
@@ -45,34 +45,59 @@ function Base(method, options) {
   this[kHandle] = private_constructor(options, {}, null, method);
   stream.Transform.$call(this, options);
 }
-Base.prototype = Object.create(stream.Transform.prototype);
-ObjectDefineProperty(Base.prototype, "_handle", {
+ZlibBase.prototype = Object.create(stream.Transform.prototype);
+ObjectDefineProperty(ZlibBase.prototype, "_handle", {
   get: function () {
     return this[kHandle];
   },
   set: function (newval) {
     //noop
   },
 });
-ObjectDefineProperty(Base.prototype, "bytesWritten", {
+ObjectDefineProperty(ZlibBase.prototype, "bytesWritten", {
   get: function () {
     return this[kHandle].bytesWritten;
   },
 });
-ObjectDefineProperty(Base.prototype, "bytesRead", {
+ObjectDefineProperty(ZlibBase.prototype, "bytesRead", {
   get: function () {
     return this[kHandle].bytesRead;
   },
 });
-ObjectDefineProperty(Base.prototype, "_closed", {
+ObjectDefineProperty(ZlibBase.prototype, "_closed", {
   get: function () {
     return this[kHandle].closed;
   },
 });
-Base.prototype.flush = function (kind, callback) {
+ObjectDefineProperty(ZlibBase.prototype, "_chunkSize", {
+  get: function () {
+    return this[kHandle].chunkSize;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_defaultFlushFlag", {
+  get: function () {
+    return this[kHandle].flush;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_finishFlushFlag", {
+  get: function () {
+    return this[kHandle].finishFlush;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_defaultFullFlushFlag", {
+  get: function () {
+    return this[kHandle].fullFlush;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_maxOutputLength", {
```

Comment:
Does this match node? Are these fields read-only and non-enumerable and non-configurable in node?

Reply:
not exactly but left a doc comment and a TODO to come back as well as made sure they were enumerable